### DISCUSSION TITLE
Setting the instance detection confidence score threshold

### DISCUSSION THREAD

**Discussion by jkbhagatio** [talmolab/sleap#1604] [Help!] [answered]
When running inference, there must somewhere be a confidence score threshold for an instance to be detected right? Let's call it "instance-detection-confidence-score-threshold". Could someone point me to where this is and how this could be adjusted prior to model export?

**Comment by roomrys** [comment, marked as answer]
Hi @jkbhagatio,

Yes! Sort-of... when loading the model, there is an argument named `peak_threshold` that allows setting a confidence threshold. However, this threshold is per point not per the entire instance. To read more on the API use of `load_model`, check out #1603. You can also set the `peak_threshold` via the [sleap-track](https://sleap.ai/guides/cli.html?highlight=cli#sleap-track) CLI:
```
  --peak_threshold PEAK_THRESHOLD
                        Minimum confidence map value to consider a peak as valid.
```

Furthermore, the peak thresholding is a parameter of inference not training, so it's more like a filter on the output of the model (not integral to the model itself).

Let us know if this helps!

Thanks,
Liezl

**Reply by talmo** [reply]
Just to add: the `peak_threshold` parameter should also affect instance detection if you're using top-down models (specifically, it applies to the centroid peak finding).

**Comment by jkbhagatio** [comment]
Thanks both! Is there a way to set this upon model export, e.g. when using `sleap-export` cli? I know now there is `--max_instances`, but still wondering if `peak_threshold` could be changed

**Reply by roomrys** [reply]
Hi @jkbhagatio,

There is no way to set `peak_threshold` when using the `sleap-export`. As mentioned above:
>peak thresholding is a parameter of inference not training, so it's more like a filter on the output of the model (not integral to the model itself)

hence why `peak_threshold` is only used as an input during inference (and not for `sleap-export`).

Thanks,
Liezl

**Reply by jkbhagatio** [reply]
Thanks @roomrys ! Is there a way to do something like `--max_instances` for id models?
